# HG changeset patch
# Parent  2c4c9adf1caa4dff43f351c757dfbdb215941222
Fix a bunch of compilation warnings in RELEASE mode

diff --git a/lifx/broadcast.c b/lifx/broadcast.c
--- a/lifx/broadcast.c
+++ b/lifx/broadcast.c
@@ -161,7 +161,8 @@
     struct sockaddr_in lifx_addr = {
         .sin_family = AF_INET,
         .sin_addr = { INADDR_BROADCAST },
-        .sin_port = htons(LGTD_LIFX_PROTOCOL_PORT)
+        .sin_port = htons(LGTD_LIFX_PROTOCOL_PORT),
+        .sin_zero = { 0 }
     };
     struct lgtd_lifx_packet_header get_pan_gateway;
     lgtd_lifx_wire_setup_header(
@@ -289,7 +290,8 @@
     struct sockaddr_in lifx_addr = {
         .sin_family = AF_INET,
         .sin_addr = { INADDR_ANY },
-        .sin_port = htons(LGTD_LIFX_PROTOCOL_PORT)
+        .sin_port = htons(LGTD_LIFX_PROTOCOL_PORT),
+        .sin_zero = { 0 }
     };
 
     err = bind(
diff --git a/lifx/gateway.c b/lifx/gateway.c
--- a/lifx/gateway.c
+++ b/lifx/gateway.c
@@ -390,6 +390,8 @@
                                      const struct lgtd_lifx_packet_header *hdr,
                                      const struct lgtd_lifx_packet_pan_gateway *pkt)
 {
+    (void)pkt;
+
     assert(gw && hdr && pkt);
 
     lgtd_debug(
diff --git a/lifx/wire_proto.c b/lifx/wire_proto.c
--- a/lifx/wire_proto.c
+++ b/lifx/wire_proto.c
@@ -308,6 +308,8 @@
 void
 lgtd_lifx_wire_decode_power_state(struct lgtd_lifx_packet_power_state *pkt)
 {
+    (void)pkt;
+
     assert(pkt);
 }
 
diff --git a/tests/core/jsonrpc/test_jsonrpc_check_and_call_power_off.c b/tests/core/jsonrpc/test_jsonrpc_check_and_call_power_off.c
--- a/tests/core/jsonrpc/test_jsonrpc_check_and_call_power_off.c
+++ b/tests/core/jsonrpc/test_jsonrpc_check_and_call_power_off.c
@@ -11,7 +11,9 @@
 lgtd_proto_power_off(struct lgtd_client *client,
                      const struct lgtd_proto_target_list *targets)
 {
-    assert(client);
+    if (!client) {
+        errx(1, "missing client!");
+    }
 
     if (strcmp(SLIST_FIRST(targets)->target, "*")) {
         errx(
diff --git a/tests/core/jsonrpc/test_jsonrpc_check_and_call_power_on.c b/tests/core/jsonrpc/test_jsonrpc_check_and_call_power_on.c
--- a/tests/core/jsonrpc/test_jsonrpc_check_and_call_power_on.c
+++ b/tests/core/jsonrpc/test_jsonrpc_check_and_call_power_on.c
@@ -11,7 +11,9 @@
 lgtd_proto_power_on(struct lgtd_client *client,
                     const struct lgtd_proto_target_list *targets)
 {
-    assert(client);
+    if (!client) {
+        errx(1, "missing client!");
+    }
 
     if (strcmp(SLIST_FIRST(targets)->target, "*")) {
         errx(
diff --git a/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_light_from_hsbk.c b/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_light_from_hsbk.c
--- a/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_light_from_hsbk.c
+++ b/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_light_from_hsbk.c
@@ -16,7 +16,9 @@
                                int kelvin,
                                int transition_msecs)
 {
-    assert(client);
+    if (!client) {
+        errx(1, "missing client!");
+    }
 
     if (strcmp(SLIST_FIRST(targets)->target, "*")) {
         errx(
diff --git a/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_light_from_hsbk_from_array.c b/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_light_from_hsbk_from_array.c
--- a/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_light_from_hsbk_from_array.c
+++ b/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_light_from_hsbk_from_array.c
@@ -16,7 +16,9 @@
                                int kelvin,
                                int transition_msecs)
 {
-    assert(client);
+    if (!client) {
+        errx(1, "missing client!");
+    }
 
     if (strcmp(SLIST_FIRST(targets)->target, "*")) {
         errx(
diff --git a/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_waveform.c b/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_waveform.c
--- a/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_waveform.c
+++ b/tests/core/jsonrpc/test_jsonrpc_check_and_call_set_waveform.c
@@ -16,7 +16,9 @@
                         int period, float cycles,
                         int skew_ratio, bool transient)
 {
-    assert(client);
+    if (!client) {
+        errx(1, "missing client");
+    }
 
     if (strcmp(SLIST_FIRST(targets)->target, "*")) {
         errx(
diff --git a/tests/core/jsonrpc/test_jsonrpc_utils.h b/tests/core/jsonrpc/test_jsonrpc_utils.h
--- a/tests/core/jsonrpc/test_jsonrpc_utils.h
+++ b/tests/core/jsonrpc/test_jsonrpc_utils.h
@@ -13,13 +13,13 @@
 void
 lgtd_proto_target_list_clear(struct lgtd_proto_target_list *targets)
 {
-    assert(targets);
+    (void)targets;
 }
 
 void
 lgtd_proto_list_tags(struct lgtd_client *client)
 {
-    assert(client);
+    (void)client;
 }
 
 #ifndef LGTD_TESTING_SET_LIGHT_FROM_HSBK
